src/qibo/models/circuit.py

```diff
@@ -1384,30 +1379,65 @@ def draw(self, line_wrap: int = 70, legend: bool = False):
 
 
 def _resolve_qubits(qubits, wire_names):
-    """Parse the input arguments for defining a circuit. Allows the user to
-    initialize the circuit as follows:
+    """Parse the input arguments for defining a circuit.
+
+    Allows the user to initialize the circuit as follows:
 
     Example:
         .. code-block:: python
             from qibo import Circuit
-            c = Circuit(3)
-            c = Circuit(3, wire_names=["q0", "q1", "q2"])
-            c = Circuit(["q0", "q1", "q2"])
-            c = Circuit(wire_names=["q0", "q1", "q2"])
+
+            circuit = Circuit(3)
+            circuit = Circuit(3, wire_names=["q0", "q1", "q2"])
+            circuit = Circuit(["q0", "q1", "q2"])
+            circuit = Circuit(wire_names=["q0", "q1", "q2"])
     """
-    if qubits is None and wire_names is not None:
-        return len(wire_names), wire_names
-    if qubits is not None and wire_names is None:
+    condition_1 = bool(qubits is None and wire_names is not None)
+    condition_2 = bool(qubits is not None and wire_names is None)
+    condition_3 = bool(qubits is not None and wire_names is not None)
```

Comment:
Of course, I perfectly understand the difficulty of finding good names. But, since you took the burden of changing this function, a bit more effort would be appreciated :P